dozer-sql/src/pipeline/product/processor.rs

```diff
@@ -48,14 +48,16 @@ impl FromProcessor {
     ) -> Result<Vec<(JoinAction, Record, Vec<u8>)>, ExecutionError> {
         let database = &self.db.ok_or(ExecutionError::InvalidDatabase)?;
 
-        self.operator.execute(
-            JoinAction::Delete,
-            from_port,
-            record,
-            database,
-            transaction,
-            reader,
-        )
+        self.operator
+            .execute(
+                JoinAction::Delete,
+                from_port,
+                record,
+                database,
+                transaction,
+                reader,
+            )
+            .map_err(|err| ExecutionError::ProductProcessorError(Box::new(err)))
```

Comment:
Do these errors have a context? Like only happening in insert etc. If so, I would do a transparent on `ExecutionError::ProductProcessorError` and specify specific situations in ProductProcessorError:Insert` etc

Not sure if this is the case.